go_backend/main.go

```diff
@@ -0,0 +1,242 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/gin-gonic/gin"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func homePage(c *gin.Context) {
+	c.JSON(200, gin.H{
+		"message": "Hello World",
+	})
+}
+
+// func postHomePage(c *gin.Context) {
+// 	body := c.Request.Body
+// 	value, err := ioutil.ReadAll(body)
+// 	if err != nil {
+// 		fmt.Println(err.Error())
+// 	}
+// 	c.JSON(200, gin.H{
+// 		"message": string(value),
+// 	})
+// }
+
+//sign up function
+func querySignUp(c *gin.Context) {
+	//getting data from request
+	// userID := c.Query("userID")
+	email := c.Query("email")
+	password := c.Query("password")
+	firstName := c.Query("firstName")
+	lastName := c.Query("lastName")
+	phoneNumber := c.Query("phoneNumber")
+	userIdentity := c.Query("userIdentity")
+
+	//connect to the db
+	db, err := sql.Open("mysql", "root:@tcp(127.0.0.1:3306)/foodapp")
```

Comment:
Connecting to DB should be in a separate file like the config file.

Reply:
do you mean move the whole part out to a config file and it can be called each time when a db connection is needed, or put it somewhere when the app starts, so the DB is always connected ?